android/src/main/java/com/polidea/flutter_ble_lib/delegate/DeviceConnectionDelegate.java

```diff
@@ -0,0 +1,196 @@
+import com.polidea.multiplatformbleadapter.RefreshGattMoment;
+import com.polidea.multiplatformbleadapter.errors.BleError;
+
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+
+public class DeviceConnectionDelegate implements MethodChannel.MethodCallHandler {
+
+    private BleAdapter bleAdapter;
+    private ConnectionStateStreamHandler streamHandler;
+
+    public DeviceConnectionDelegate(BleAdapter bleAdapter, ConnectionStateStreamHandler streamHandler) {
+        this.bleAdapter = bleAdapter;
+        this.streamHandler = streamHandler;
+    }
+
+    @Override
+    public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
+        String deviceId = call.argument(ArgumentKey.DEVICE_IDENTIFIER);
+        switch (call.method) {
+            case MethodName.CONNECT_TO_DEVICE:
+                connectToDevice(deviceId,
+                        call.<Boolean>argument(ArgumentKey.IS_AUTO_CONNECT),
+                        call.<Integer>argument(ArgumentKey.REQUEST_MTU),
+                        call.<Boolean>argument(ArgumentKey.REFRESH_GATT),
+                        call.<Long>argument(ArgumentKey.TIMEOUT_MILLIS),
+                        result);
+                return;
+            case MethodName.IS_DEVICE_CONNECTED:
+                isDeviceConnected(deviceId, result);
+                return;
+            case MethodName.OBSERVE_CONNECTION_STATE:
+                observeConnectionState(deviceId, call.<Boolean>argument(ArgumentKey.EMIT_CURRENT_VALUE), result);
+                return;
+            case MethodName.CANCEL_CONNECTION:
+                cancelConnection(deviceId, result);
+                return;
+            default:
+                throw new IllegalStateException("");
+        }
+    }
+
+    private void connectToDevice(String deviceId, Boolean isAutoConnect, Integer requestMtu, Boolean refreshGatt, Long timeoutMillis, @NonNull final MethodChannel.Result result) {
+        RefreshGattMoment refreshGattMoment = null;
+        if (refreshGatt) refreshGattMoment = RefreshGattMoment.ON_CONNECTED;
+
+        final SafeMainThreadResolver safeMainThreadResolver = new SafeMainThreadResolver<>(
+                new OnSuccessCallback<Object>() {
+                    @Override
+                    public void onSuccess(Object data) {
+                        result.success(null);
+                    }
+                },
+                new OnErrorCallback() {
+                    @Override
+                    public void onError(BleError error) {
+                        result.error(String.valueOf(error.errorCode.code), error.reason, null); //TODO @BartoszWilk convert this error
+                    }
+                });
+
+        bleAdapter.connectToDevice(deviceId,
+                new ConnectionOptions(isAutoConnect, requestMtu, refreshGattMoment, timeoutMillis, 0 /* BluetoothGatt.CONNECTION_PRIORITY_BALANCED */),
```

Comment:
it would be great to have `0` extracted to a const instead putting the info to a comment

Reply:
👍